test/OpenFeature.E2ETests/Steps/HooksStepDefinitions.cs

```diff
@@ -0,0 +1,219 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenFeature.Model;
+using OpenFeature.Providers.Memory;
+using Reqnroll;
+using Xunit;
+
+namespace OpenFeature.E2ETests.Steps;
+
+[Binding]
+[Scope(Feature = "Evaluation details through hooks")]
+public class HooksStepDefinitions
+{
+    private FeatureClient? _client;
+    private TestHook? _testHook;
+
+    private static readonly IDictionary<string, Flag> E2EFlagConfig = new Dictionary<string, Flag>
+    {
+        {
+            "boolean-flag", new Flag<bool>(
+                variants: new Dictionary<string, bool> { { "on", true }, { "off", false } },
+                defaultVariant: "on"
+            )
+        }
+    };
+
+    [Given(@"a stable provider")]
+    public void GivenAStableProvider()
+    {
+        var memProvider = new InMemoryProvider(E2EFlagConfig);
+        Api.Instance.SetProviderAsync(memProvider).Wait();
+        this._client = Api.Instance.GetClient("TestClient", "1.0.0");
+    }
+
+    [Given(@"a client with added hook")]
+    public void GivenAClientWithAddedHook()
+    {
+        this._testHook = new TestHook();
+        this._client!.AddHooks(this._testHook);
+    }
+
+    [Given(@"a boolean-flag with key ""(.*)"" and a default value ""(.*)""")]
+    public async Task GivenABoolean_FlagWithKeyAndADefaultValue(string key, string defaultValue)
+    {
+        _ = await this._client!.GetBooleanValueAsync(key, bool.Parse(defaultValue)).ConfigureAwait(false);
```

Comment:
i recommend to persist the key, value and type here into variables, and not do any verification. 

The given steps should setup the test, not test already functionality we want to test. 
The when steps should execute.
The then steps should verify.

So ideally, in the `GivenABoolean_FlagWithKeyAndADefaultValue` you persist the name, value and type of the flag.
Within `WhenTheFlagWasEvaluatedWithDetails` you call the client for the details.
Within a then, you verify the data.

Reply:
This should now be fixed and part of `test/OpenFeature.E2ETests/Steps/BaseStepDefinitions.cs`.